### PR TITLE
Create app

### PR DESCRIPTION
**Pull request by ribeirojose**
I used Flask to integrate the project into an API that is fetched by a React app on the client's side.

### CODE REVIEW

File: app/client/App.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    if (!this.state.data) {
+      return <div>Loading...</div>
+    }
+
+    return (
+      <div>
+        {this.state.data.map((day, idx) => <Table name={idx} data={day}></Table>)}
+      </div>
+    )
```
Comment (pkretzschmar) [inline app/client/App.js]: Try to use something like that inside your return method, its gonna be easier to understand. 
```
<div> 
    {!this.state.data ? (
    "Loading...
    ) : (
    this.state.data.map(day, idx) => <Table name={idx} data={day}} />
</div>`
```

File: app/client/Table.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+class Table extends Component {
+  constructor(props) {
+    super(props) //since we are extending class Table so we have to use super in order to override Component class constructor
+    this.hours = Object.keys(this.props.data)
+    this.days = ["Mon", "Tue", "Wed", "Thu", "Fri"]
+  }
+
+  renderTableRow() {
+    let rows = this.hours.map((hour) => [...[hour], ...Object.values(this.props.data[hour])])
+    return rows.map((row, i) => <tr key={i}>{row.map((el, idx) => <td key={idx}>{el}</td>)}</tr>)
```
Comment (pkretzschmar) [inline app/client/Table.js]: Try to alocate this in a variable before u return, them you can tell to the others dev's what is it easily what this suppose to be/do using a good name for the var!

**Review by pkretzschmar — COMMENTED** [review]
Nice work man! Just some adjustments and it'll be perfect!

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>DCS Editor</title>
```
Comment (lukasab) [inline public/index.html]: `<title>Desk Allocation for NEO</title>`

**Review by lukasab — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+node_js:
+  - 8
+
+script:
+  - npm run build
+
+branches:
+  only:
+  - gh-pages
+  - /.*/
```
[the remark this answers is not in the document]
Comment (ribeirojose) [reply .travis.yml]: Not for now, I'll remove this file =)

File: package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    "eslint": "^5.0.0",
+    "eslint-config-airbnb": "^17.0.0",
+    "eslint-plugin-import": "^2.11.0",
+    "eslint-plugin-jsx-a11y": "^6.0.3",
+    "eslint-plugin-react": "^7.7.0",
+    "file-loader": "^3.0.0",
+    "html-webpack-plugin": "^3.2.0",
+    "style-loader": "^0.23.0",
+    "url-loader": "^1.0.1",
+    "webpack": "^4.5.0",
```
Comment (lukasab) [inline package.json]: shouldn't webpack be in normal dependencies? For the build script

**Review by lukasab — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (ribeirojose) [reply package.json]: No, it is only used for bundling the client app, but it's not used by the app in production.  There's a _very_ discussion on this [here](https://github.com/webpack/webpack/issues/520)

**Pull request merged**